crates/db_schema/src/impls/community.rs

```diff
@@ -357,14 +344,34 @@ impl CommunityFollower {
     .then_some(())
     .ok_or(LemmyErrorType::CommunityHasNoFollowers.into())
   }
+
+  pub async fn approve(
+    pool: &mut DbPool<'_>,
+    community_id: CommunityId,
+    follower_id: PersonId,
+    approved_by_id: PersonId,
+  ) -> LemmyResult<()> {
+    let conn = &mut get_conn(pool).await?;
+    diesel::update(community_follower::table.find((follower_id, community_id)))
+      .set((
+        community_follower::state.eq(CommunityFollowerState::Accepted),
+        community_follower::approved_by.eq(approved_by_id),
+      ))
+      .get_result::<Self>(conn)
+      .await?;
+    Ok(())
+  }
 }
 
-impl Queryable<sql_types::Nullable<sql_types::Bool>, Pg> for SubscribedType {
-  type Row = Option<bool>;
+impl Queryable<sql_types::Nullable<crate::schema::sql_types::CommunityFollowerState>, Pg>
```

Comment:
I don't get this one either. Its easy enough to create a function that maps from a return communityfollowerstate -> subscribedtype. That can easily be done after the query is returned, and this isn't necessary.

Reply:
Again it needs to map from None to SubscribedType::NotSubscribed which doesnt work with a simple select.